frontend/src/hex/HexView.svelte

```diff
@@ -265,123 +267,137 @@
     return info;
   }
 
-  let dataSearchResults = {};
-
-  // React to local data searches
-  $: {
-    const localDataSearchResults = dataSearchResults;
-
-    if (
-      localDataSearchResults.matches?.length > 0 &&
-      (localDataSearchResults.index || localDataSearchResults.index === 0)
-    ) {
-      dataLenPromise.then((dataLength) => {
-        $scrollY.top =
-          localDataSearchResults.matches[localDataSearchResults.index][0] /
-          dataLength;
-      });
-    }
+  function resetResource() {
+    dataSearchResults.matches = undefined;
+    currentPosition = 0;
   }
 
-  async function searchHex(query, options) {
-    return await $selectedResource.search_data(query, options);
+  function refreshHeight() {
+    hexDisplay = document.getElementById("scrollable");
+    $screenHeight =
+      Math.floor(hexDisplay.offsetHeight / lineHeight) * alignment;
+    chunkDataPromise = dataLenPromise.then(getNewData);
   }
+
+  onMount(() => {
+    hexDisplay = document.getElementById("scrollable");
+    refreshHeight();
+  });
```

Comment:
The more "correct" way to do this in Svelte is to `bind:this` on the `scrollable` element. The variable should be reactively updated on mount.